### PR TITLE
Collect logs for Windows Nodes

### PR DESCRIPTION
**Pull request by haofan-ms**
Modify the getkuberneteslogs.sh to collect logs for windows nodes as well.

### CODE REVIEW

File: diagnosis/README.md [hunk 1]
```diff
@@ -36,4 +46,6 @@ The logs retrieved by `getkuberneteslogs.sh` are the following:
 `--api-model`                  - Persists apimodel.json file in an Azure Stack Storage account. 
                                  Upload apimodel.json file to storage account happens when `--upload-logs` parameter is also provided.
 
+`--include-windows-nodes`      - Collect log for Windows nodes. Only apply to scenarios when Windows nodepools exist in the Kubernetes clusters.
```
Comment (jadarsie) [inline diagnosis/README.md:49]: I would try to infer this instead of asking the user to set the flag

Comment (haofan-ms) [reply diagnosis/README.md]: addressed

File: diagnosis/README.md [hunk 2]
```diff
@@ -19,6 +19,16 @@ The logs retrieved by `getkuberneteslogs.sh` are the following:
...
 
+Some additional logs are retrieved for windows nodes:
+
+ - Log file `c:\Azure\CustomDataSetupScript.log`
+ - kube-proxy status and journal
+ - containerd status and journal
+ - azure-vnet log and azure-vnet-telemetry log
+ - ETW events for docker
+ - ETW events for Hyper-V
+ - ETW events for some
```
Comment (jadarsie) [inline diagnosis/README.md:30]: this is missing something

File: diagnosis/getkuberneteslogs.sh [hunk 3]
```diff
@@ -130,6 +130,17 @@ processDvmHost()
     ssh ${SSH_FLAGS} ${USER}@${host} "rm -f collectlogs.sh ${DVM_NAME}.zip"
 }
 
+processWindowsHost()
+{
+    host=$1
+
+    echo "[$(date +%Y%m%d%H%M%S)][INFO] Processing windows-host ${host}"
+    scp ${SCP_FLAGS} -o ProxyCommand="${PROXY_CMD}" collect-windows-logs.ps1 ${USER}@${host}:"C:/k/debug/collect-windows-logs.ps1"
```
Comment (jadarsie) [inline diagnosis/getkuberneteslogs.sh:138]: Will this will override the windows log collection script?

Comment (jadarsie) [reply diagnosis/getkuberneteslogs.sh:138]: I think it will conflict with the script shipped with the aks windows vhd

File: diagnosis/README.md [hunk 4]
```diff
@@ -19,6 +19,16 @@ The logs retrieved by `getkuberneteslogs.sh` are the following:
...
 
+Some additional logs are retrieved for windows nodes:
+
+ - Log file `c:\Azure\CustomDataSetupScript.log`
+ - kube-proxy status and journal
+ - containerd status and journal
+ - azure-vnet log and azure-vnet-telemetry log
+ - ETW events for docker
+ - ETW events for Hyper-V
+ - ETW events for some system events
```
Comment (jadarsie) [inline diagnosis/README.md:30]: I would rather remove the this line if it does not include a specific detail

File: diagnosis/README.md [hunk 5]
```diff
@@ -19,6 +19,16 @@ The logs retrieved by `getkuberneteslogs.sh` are the following:
 - docker status and journal
 - kube-system snapshot
 
+Some additional logs are retrieved for windows nodes:
```
Comment (jadarsie) [inline diagnosis/README.md:22]: `Windows`

File: diagnosis/getkuberneteslogs.sh [hunk 6]
```diff
@@ -293,11 +304,24 @@ then
...
     do
         processHost ${host}
     done
+
+    #Get Windoews nodes log if Windows nodes exist
+    WINDOWS_NODES=$(az vm list -g ${RESOURCE_GROUP} --show-details --query "[*].{Name:name,ip:privateIps}" --output tsv | grep -v 'k8s-\|vmd-' | cut -f 1)
+
+    if [ -n "$WINDOWS_NODES" ]
+    then
+        echo "[INFO] Windows nodes detected, will collect log for windows nodes."
```
Comment (jadarsie) [inline diagnosis/getkuberneteslogs.sh:318]: I would remove this, not consistent with the linux nodes block

File: diagnosis/README.md (hunk 4 above)
Comment (haofan-ms) [reply diagnosis/README.md]: Removed

File: diagnosis/README.md (hunk 5 above)
Comment (haofan-ms) [reply diagnosis/README.md]: Updated

File: diagnosis/getkuberneteslogs.sh (hunk 6 above)
Comment (haofan-ms) [reply diagnosis/getkuberneteslogs.sh]: Updated

**Pull request merged**